src/otx/utils/utils.py

```diff
@@ -114,3 +126,28 @@ def remove_matched_files(directory: Path, pattern: str, file_to_leave: Path | No
     for weight in directory.rglob(pattern):
         if weight != file_to_leave:
             weight.unlink()
+
+
+def is_xpu_available() -> bool:
+    """Checks if XPU device is available."""
+    global XPU_AVAILABLE  # noqa: PLW0603
+    if XPU_AVAILABLE is None:
+        XPU_AVAILABLE = hasattr(torch, "xpu") and torch.xpu.is_available()
+    return XPU_AVAILABLE
+
+
+def patch_packages_xpu(task: str | OTXTaskType, accelerator: str | DeviceType) -> None:
+    """Patch packages when xpu is available."""
+    if accelerator == "xpu" and ("DETECTION" in task or "INSTANCE_SEGMENTATION" in task):
+        import numpy as np
+        from mmcv.ops.nms import NMSop
+        from mmcv.ops.roi_align import RoIAlign
+        from mmengine.structures import instance_data
+
+        from otx.algo.detection.utils import monkey_patched_nms, monkey_patched_roi_align
+
+        long_type_tensor = Union[torch.LongTensor, torch.xpu.LongTensor]
+        bool_type_tensor = Union[torch.BoolTensor, torch.xpu.BoolTensor]
+        instance_data.IndexType = Union[str, slice, int, list, long_type_tensor, bool_type_tensor, np.ndarray]
+        NMSop.forward = monkey_patched_nms
+        RoIAlign.forward = monkey_patched_roi_align
```

Comment:
```suggestion
        global _nms_op_forward, _roi_align_forward

        _nms_op_forward = NMSop.forward
        _roi_align_forward = RoIAlign.forward
        NMSop.forward = monkey_patched_nms
        RoIAlign.forward = monkey_patched_roi_align

def revert_packages_xpu():
    NMSop.forward = _nms_op_forward
    RoIAlign.forward = _roi_align_forward
    
class XPUAccelerator
....
    def teardown():
        revert_packages_xpu()
```

Reply:
(Done)[https://github.com/openvinotoolkit/training_extensions/pull/3094/commits/8fb52df736f8c03245d64994bbadd54debab8007]. I moved it to accelerator to avoid working with global variables